pylint/checkers/typecheck.py

```diff
@@ -1035,6 +1053,25 @@ def visit_assign(self, node):
             else:
                 self.add_message("assignment-from-none", node=node)
 
+    def _check_dundername_is_string(self, node):
+        """
+        Check a string is assigned to self.__name__
+        """
+
+        # Check the left hand side of the assignment is <something>.__name__
+        lhs = list(node.get_children())[0]
+        if not isinstance(lhs, astroid.node_classes.AssignAttr):
+            return
+        if not lhs.attrname == "__name__":
+            return
+
+        # If the right hand side is not a string
+        rhs = list(node.get_children())[1]
```

Comment:
That will be `node.value` instead.

Reply:
Thanks! Should be fixed in https://github.com/PyCQA/pylint/pull/3271/commits/abbcafab1f23b72cfc37af10861d3ded58dd7a6d